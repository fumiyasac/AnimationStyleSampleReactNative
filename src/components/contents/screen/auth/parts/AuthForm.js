/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 * @flow
 */

import React, { Component } from 'react';

import { 
  Form,
  Item,
  Input,
  Label
} from 'native-base';

// 認証コンテンツ用の共通コンポーネント
import AuthFormIntro from './AuthFormIntro';
import AuthFormErrorMessage from './AuthFormErrorMessage';
import AuthFormSubmitButton from './AuthFormSubmitButton';

export default class AuthForm extends React.Component {
  render() {
    return (
      <Form>
        <AuthFormIntro />
        <Item stackedLabel last>
          <Label>ユーザー名:</Label>
          <Input />
        </Item>
        <Item stackedLabel last>
          <Label>パスワード:</Label>
          <Input />
        </Item>
        <AuthFormErrorMessage />
        <AuthFormSubmitButton />
      </Form>
    );
  };
}
