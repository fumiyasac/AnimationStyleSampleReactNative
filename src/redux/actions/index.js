/**
 * actions/index.js
 */

export * from './AuthActions';
export * from './FeedActions';
export * from './NewsActions';
