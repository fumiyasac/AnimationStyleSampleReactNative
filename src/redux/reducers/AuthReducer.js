/**
 * reducers/AuthReducer.js
 *
 * 認証状態を管理するためのReducer
 * → 定義されたaction経由で実行されるステートに関する処理に関するロジックを定義する
 */

// 認証関連のアクションタイプ定義のインポート宣言
import { 
  EMAIL_CHANGED,
  PASSWORD_CHANGED,
  LOGIN_USER_SUCCESS,
  LOGIN_USER_FAILURE,
  LOGIN_USER_INVALID,
  LOGIN_USER
} from '../actions/types';

// 初期状態のステート定義（オブジェクトの形にする）
const initialState = { email: '', password: '', user: null, error: '', loading: false };

// 選択されたケースを元にstateの更新を行うメソッド（アクションのタイプに応じての場合分けがされている）
export default (state = initialState, action) => {
  switch (action.type) {
    case EMAIL_CHANGED:
      return { ...state, email: action.payload };
    case PASSWORD_CHANGED:
      return { ...state, password: action.payload };
    case LOGIN_USER_SUCCESS:
      return { ...state, ...initialState, user: action.payload };
    case LOGIN_USER_FAILURE:
      return { ...state, error: '認証に失敗しました。', password: '', loading: false };
    case LOGIN_USER_INVALID:
      return { ...state, error: '入力項目の形式が正しくありません。', loading: false };
    case LOGIN_USER:
      return { ...state, loading: true, error: '' };
    default:
      return state;
  }
};