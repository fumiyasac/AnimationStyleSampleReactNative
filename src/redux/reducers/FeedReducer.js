/**
 * reducers/FeedReducer.js
 */

import {
  FEED_FETCH,
  FEED_FETCH_SUCCESS,
  FEED_FETCH_FAILURE 
} from '../actions/types';

// 初期状態のステート定義（オブジェクトの形にする）
const initialState = { feed: null, error: '', loading: false };

// 選択されたケースを元にstateの更新を行うメソッド
export default (state = initialState, action) => {
  switch (action.type) {
    case FEED_FETCH:
      return { ...state, loading: true, error: '' };
    case FEED_FETCH_SUCCESS:
      return { ...state, feed: action.payload };
    case FEED_FETCH_FAILURE:
      return { ...state, error: 'ニュース情報の取得に失敗しました。', loading: false };
    default:
      return state;
  }
};