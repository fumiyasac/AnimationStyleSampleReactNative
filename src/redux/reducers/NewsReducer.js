/**
 * reducers/NewsReducer.js
 */

import {
  NEWS_FETCH,
  NEWS_FETCH_SUCCESS,
  NEWS_FETCH_FAILURE 
} from '../actions/types';

// 初期状態のステート定義（オブジェクトの形にする）
const initialState = { newsList: null, error: '', loading: false };

// 選択されたケースを元にstateの更新を行うメソッド
export default (state = initialState, action) => {
  switch (action.type) {
    case NEWS_FETCH:
      return { ...state, ...initialState, loading: true };
    case NEWS_FETCH_SUCCESS:
      return { ...state, ...initialState, newsList: action.payload };
    case NEWS_FETCH_FAILURE:
      return { ...state, error: '新着お知らせの取得に失敗しました。', loading: false };
    default:
      return state;
  }
};