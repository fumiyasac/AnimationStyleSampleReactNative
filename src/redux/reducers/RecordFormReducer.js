/**
 * reducers/RecordFormReducer.js
 */

// 記録変更関連のアクションタイプ定義のインポート宣言
import { 
  RECORD_REFRESH,
  RECORD_UID_CHANGED,
  RECORD_TITLE_CHANGED,
  RECORD_CONTENT_CHANGED,
  RECORD_REQUEST_INVALID,
  RECORD_REQUEST_CREATE,
  RECORD_REQUEST_UPDATE,
  RECORD_REQUEST_DELETE,
  RECORD_REQUEST_SUCCESS,
  RECORD_REQUEST_FAILURE,
} from '../actions/types';

// 初期状態のステート定義（オブジェクトの形にする）
const initialState = { uid: null, title: '', content: '', error: '', loading: false };

// 選択されたケースを元にstateの更新を行うメソッド（アクションのタイプに応じての場合分けがされている）
export default (state = initialState, action) => {
  switch (action.type) {
    case RECORD_UID_CHANGED:
      return { ...state, uid: action.payload };      
    case RECORD_TITLE_CHANGED:
      return { ...state, title: action.payload };
    case RECORD_CONTENT_CHANGED:
      return { ...state, content: action.payload };
    case RECORD_REQUEST_CREATE:
    case RECORD_REQUEST_UPDATE:
    case RECORD_REQUEST_DELETE:
      return { ...state, loading: true, error: '' };
    case RECORD_REQUEST_SUCCESS:
    case RECORD_REFRESH:
    return initialState;
    case RECORD_REQUEST_INVALID:
      return { ...state, error: 'タイトルまたは本文が入力されていません。', loading: false };
    case RECORD_REQUEST_FAILURE:
      return { ...state, error: '記録データの更新に失敗しました。', loading: false };
    default:
      return state;
  }
};