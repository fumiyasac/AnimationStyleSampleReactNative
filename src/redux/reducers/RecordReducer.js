/**
 * reducers/RecordReducer.js
 */

import { 
  RECORD_FETCH,
  RECORD_FETCH_SUCCESS
} from '../actions/types';

// 初期状態のステート定義（オブジェクトの形にする）
const initialState = { recordList: null, loading: false };;

// 選択されたケースを元にstateの更新を行うメソッド
export default (state = initialState, action) => {
  switch (action.type) {
    case RECORD_FETCH:
      return { ...state, ...initialState, loading: true };
    case RECORD_FETCH_SUCCESS:
      return { ...state, ...initialState, recordList: action.payload };
    default:
      return state;
  }
};
