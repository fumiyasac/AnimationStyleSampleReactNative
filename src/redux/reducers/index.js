/**
 * reducers/index.js
 * 
 * データ表示で使用するReducerの定義（アプリケーションのステート一覧に関するもの）
 * Reduxの原則の1つであるグローバルな状態ツリーがすべてのソースとなるが、一部を更新するような形がほとんどなので役割に応じてReducerを分割する。
 */

import { combineReducers } from 'redux';

import AuthReducer from './AuthReducer';
import NewsReducer from './NewsReducer';
import FeedReducer from './FeedReducer';
import RecordReducer from './RecordReducer';
import RecordFormReducer from './RecordFormReducer';

export default combineReducers({
  auth: AuthReducer,
  feed: FeedReducer,
  news: NewsReducer,
  records: RecordReducer,
  recordForm: RecordFormReducer
});