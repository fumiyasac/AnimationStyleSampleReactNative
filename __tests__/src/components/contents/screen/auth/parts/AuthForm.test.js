import 'react-native';
import React from 'react';

import AuthForm from '../../../../../../../src/components/contents/screen/auth/parts/AuthForm';

import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import ReduxThunk from 'redux-thunk';
import logger from 'redux-logger';
import reducers from '../../../../../../../src/redux/reducers';

import renderer from 'react-test-renderer';

it('AuthFormコンポーネントが正しく表示されること', () => {
  const store = createStore(reducers, {}, applyMiddleware(ReduxThunk, logger));
  const tree = renderer.create(
    <Provider store={store}>
      <AuthForm />
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
