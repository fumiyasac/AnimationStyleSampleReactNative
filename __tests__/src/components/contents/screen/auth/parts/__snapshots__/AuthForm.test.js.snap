// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthFormコンポーネントが正しく表示されること 1`] = `
<View
  style={Object {}}
>
  <View>
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          },
          Object {
            "fontSize": 14,
            "fontWeight": "bold",
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 14,
          },
        ]
      }
      uppercase={false}
    >
      Firebase + NativeBase Sample
    </Text>
    <Text
      note={true}
      style={
        Array [
          Object {
            "color": "#a7a7a7",
            "fontFamily": "System",
            "fontSize": 14,
          },
          Object {
            "fontSize": 13,
            "padding": 10,
          },
        ]
      }
      uppercase={false}
    >
      こちらはFirebaseとNativeBaseを組み合わせて作成したサンプルの事例になります。※お手元で動かす場合は、お使いの際にはご自身のFirebaseアカウント情報をご利用下さい。
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderBottomWidth": 1,
        "borderColor": "#D9D5DC",
        "borderLeftWidth": 0,
        "borderRadius": undefined,
        "borderRightWidth": 0,
        "borderTopWidth": 0,
        "borderWidth": undefined,
        "flexDirection": null,
        "marginLeft": 0,
        "minHeight": 65,
        "opacity": 1,
        "paddingLeft": 15,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "flex-start",
            "color": "#575757",
            "fontSize": 15,
            "left": null,
            "paddingLeft": null,
            "paddingRight": 5,
            "paddingTop": 5,
            "position": null,
            "right": null,
            "top": 5,
          },
          Object {
            "color": "#333333",
            "fontSize": 13,
            "fontWeight": "bold",
          },
        ]
      }
    >
      メールアドレス:
    </Text>
    <TextInput
      allowFontScaling={true}
      editable={true}
      onChangeText={[Function]}
      placeholder="例）username@example.com"
      placeholderTextColor="#575757"
      style={
        Array [
          Object {
            "alignSelf": "stretch",
            "color": "#000",
            "flex": 1,
            "fontSize": 17,
            "height": 50,
            "lineHeight": 18,
            "marginLeft": null,
            "paddingLeft": null,
            "paddingRight": 5,
            "top": 1.5,
            "width": null,
          },
          Object {
            "color": "#666666",
            "fontSize": 13,
          },
        ]
      }
      underlineColorAndroid="rgba(0,0,0,0)"
      value=""
    />
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderBottomWidth": 1,
        "borderColor": "#D9D5DC",
        "borderLeftWidth": 0,
        "borderRadius": undefined,
        "borderRightWidth": 0,
        "borderTopWidth": 0,
        "borderWidth": undefined,
        "flexDirection": null,
        "marginLeft": 0,
        "minHeight": 65,
        "opacity": 1,
        "paddingLeft": 15,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "flex-start",
            "color": "#575757",
            "fontSize": 15,
            "left": null,
            "paddingLeft": null,
            "paddingRight": 5,
            "paddingTop": 5,
            "position": null,
            "right": null,
            "top": 5,
          },
          Object {
            "color": "#333333",
            "fontSize": 13,
            "fontWeight": "bold",
          },
        ]
      }
    >
      パスワード:
    </Text>
    <TextInput
      allowFontScaling={true}
      editable={true}
      onChangeText={[Function]}
      placeholder="例）●●●●●●●●"
      placeholderTextColor="#575757"
      secureTextEntry={true}
      style={
        Array [
          Object {
            "alignSelf": "stretch",
            "color": "#000",
            "flex": 1,
            "fontSize": 13,
            "height": 50,
            "lineHeight": 18,
            "marginLeft": null,
            "paddingLeft": null,
            "paddingRight": 5,
            "top": 1.5,
            "width": null,
          },
          Object {
            "color": "#666666",
            "fontSize": 13,
          },
        ]
      }
      underlineColorAndroid="rgba(0,0,0,0)"
      value=""
    />
  </View>
  <View>
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          },
          Object {
            "alignSelf": "center",
            "color": "#ff0000",
            "fontSize": 13,
            "paddingBottom": 14,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 14,
          },
        ]
      }
      uppercase={false}
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "paddingBottom": 15,
        "paddingLeft": 10,
        "paddingRight": 10,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#4169e1",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 2,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": undefined,
          "shadowOffset": undefined,
          "shadowOpacity": undefined,
          "shadowRadius": undefined,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#fff",
              "fontFamily": "System",
              "fontSize": 16.5,
              "marginLeft": 0,
              "marginRight": 0,
              "paddingLeft": 16,
              "paddingRight": 16,
            },
            Object {
              "alignSelf": "center",
              "backgroundColor": "#4169e1",
              "color": "#ffffff",
              "fontSize": 13,
              "paddingBottom": 8,
              "paddingTop": 8,
            },
          ]
        }
        uppercase={false}
      >
        既存ユーザー情報でログインする
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "paddingBottom": 15,
        "paddingLeft": 10,
        "paddingRight": 10,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#4169e1",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 2,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": undefined,
          "shadowOffset": undefined,
          "shadowOpacity": undefined,
          "shadowRadius": undefined,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#fff",
              "fontFamily": "System",
              "fontSize": 16.5,
              "marginLeft": 0,
              "marginRight": 0,
              "paddingLeft": 16,
              "paddingRight": 16,
            },
            Object {
              "alignSelf": "center",
              "backgroundColor": "#4169e1",
              "color": "#ffffff",
              "fontSize": 13,
              "paddingBottom": 8,
              "paddingTop": 8,
            },
          ]
        }
        uppercase={false}
      >
        新規ユーザー情報を登録する
      </Text>
    </View>
  </View>
</View>
`;
